### PR TITLE
Build process

### LINKED ISSUE #8: Develop a repeatable way to cut releases that include compiled binaries

### PR DESCRIPTION
**Pull request by etsauer**
This pull request adds a Makefile which enables us to automatically build and release the project. A sample release can be seen on [my fork](https://github.com/etsauer/dash/releases/tag/v0.0.1). The Makefile also included a linting target, which forced me to make some code changes to comply with the linting. Also added an open source license (copied from applier)

resolves #8

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,83 @@
+LAST_TAG := $(shell git describe --abbrev=0 --tags)
+
+USER := etsauer
```
Comment (tylerauerbeck) [inline Makefile]: Do we want to hardcode your username in here?

**Review by tylerauerbeck — COMMENTED** [review]
@etsauer Just one note about the Makefile, but otherwise everything looks good.

**Comment by etsauer** [comment]
@tylerauerbeck I think this is ready to go now.

**Review by tylerauerbeck — APPROVED** [review]
LGTM.

**Pull request merged**